src/main/java/com/minecolonies/coremod/colony/buildings/workerbuildings/BuildingDyer.java

```diff
@@ -196,6 +208,68 @@ public void requestUpgrade(final PlayerEntity player, final BlockPos builder)
         super.requestUpgrade(player, builder);
     }
 
+    @Override
+    public IRecipeStorage getFirstRecipe(Predicate<ItemStack> stackPredicate)
+    {
+        IRecipeStorage recipe = super.getFirstRecipe(stackPredicate);
+
+        if(recipe == null && stackPredicate.test(new ItemStack(Items.WHITE_WOOL)))
+        {
+            final ResourceLocation wool = new ResourceLocation("minecraft", "wool");
+            final HashMap<ItemStorage, Integer> inventoryCounts = new HashMap<>();
+
+            final Set<IBuilding> wareHouses = colony.getBuildingManager().getBuildings().values().stream()
+                                                    .filter(building -> building instanceof BuildingWareHouse)
+                                                    .collect(Collectors.toSet());
+            
+            final List<ItemStorage> woolItems = ItemTags.getCollection().getOrCreate(wool).getAllElements().stream()
```

Comment:
I think we should wait for the multithreaded simulation solution before doing this. This can get extremely expensive. If this is executed every call. The same thing for the "inWarehouse" general solution. I think we should wait for the threaded solution first before we take this step.

Reply:
It's actually pretty fast. In the dyer case it ONLY queries the warehouse when trying to produce white wool. And I just submitted a change to only query the warehouse in the general case when there are duplicate output recipes.